File: sdk/nodejs/src/exceptions.ts
/**
 * 异常类
 */

export class InkPathError extends Error {
  constructor(message: string) {
    super(message);
    this.name = 'InkPathError';
  }
}

export class APIError extends InkPathError {
  code: string;
  statusCode: number;

  constructor(message: string, code: string = 'API_ERROR', statusCode: number = 0) {
    super(`[${code}] ${message}`);
    this.name = 'APIError';
    this.code = code;
    this.statusCode = statusCode;
  }
}

export class ValidationError extends InkPathError {
  code: string;

  constructor(message: string, code: string = 'VALIDATION_ERROR') {
    super(`[${code}] ${message}`);
    this.name = 'ValidationError';
    this.code = code;
  }
}

File: sdk/python/inkpath/exceptions.py
"""
InkPath SDK异常类
"""


class InkPathError(Exception):
    """InkPath SDK基础异常"""
    pass


class APIError(InkPathError):
    """API调用错误"""
    def __init__(self, message: str, code: str = "API_ERROR", status_code: int = 0):
        self.message = message
        self.code = code
        self.status_code = status_code
        super().__init__(f"[{code}] {message}")


class ValidationError(InkPathError):
    """请求验证错误（422）"""
    def __init__(self, message: str, code: str = "VALIDATION_ERROR"):
        self.message = message
        self.code = code
        super().__init__(f"[{code}] {message}")
